crates/sui/src/keytool.rs

```diff
@@ -878,53 +923,80 @@ impl KeyToolCommand {
-                        if tx_bytes.is_none() || provider.is_none() || curr_epoch.is_none() {
+                        if bytes.is_none() || curr_epoch.is_none() {
                             return Ok(CommandOutput::ZkLoginSigVerify(ZkLoginSigVerifyResponse {
-                                tx_data: None,
+                                data: None,
                                 parsed: Some(serde_json::to_string(&zk)?),
                                 res: None,
                                 jwks: None,
                             }));
                         }
 
-                        let tx_data: TransactionData = bcs::from_bytes(
-                            &Base64::decode(&tx_bytes.unwrap())
-                                .map_err(|e| anyhow!("Invalid base64 tx data: {:?}", e))?,
-                        )?;
                         let client = reqwest::Client::new();
-                        let jwks = fetch_jwks(
-                            &OIDCProvider::from_str(&provider.clone().unwrap())
-                                .map_err(|_| anyhow!("Unsupported provider:"))?,
-                            &client,
-                        )
-                        .await?;
+                        let mut jwks = Vec::new();
+                        for p in [
+                            OIDCProvider::Google,
+                            OIDCProvider::Twitch,
+                            OIDCProvider::Facebook,
+                            OIDCProvider::Slack,
+                            OIDCProvider::Apple,
+                            OIDCProvider::Kakao,
+                        ] {
+                            let mut e = fetch_jwks(&p, &client).await?;
+                            jwks.append(&mut e);
+                        }
                         let parsed: ImHashMap<JwkId, JWK> = jwks.clone().into_iter().collect();
-                        let aux_verify_data = VerifyParams::new(
-                            parsed,
-                            vec![
-                                OIDCProvider::Facebook,
-                                OIDCProvider::Twitch,
-                                OIDCProvider::Google,
-                            ],
-                            ZkLoginEnv::Prod,
-                            true,
-                        );
-                        let res = zk.verify_authenticator(
-                            &IntentMessage::new(Intent::sui_transaction(), tx_data.clone()),
-                            tx_data.execution_parts().1,
-                            Some(curr_epoch.unwrap()),
-                            &aux_verify_data,
-                        );
+                        let env = match network.as_str() {
+                            "devnet" | "localnet" => ZkLoginEnv::Test,
+                            "mainnet" | "testnet" => ZkLoginEnv::Prod,
+                            _ => return Err(anyhow!("Invalid network")),
```

Comment:
Don't know a lot about ZKLogin and perhaps I am lacking context, but wouldn't it be possible for some folks to have their own network set up and aliased differently than devnet / localnet / testnet / mainnet? So I guess if someone sets their own local network, it should always be called `localnet`.

Reply:
there are only four networks for sui.

Reply:
Ok, I've been looking more through the code. I think passing the `network` as an argument is not the best (I saw it in other commands too), but it defeats the purpose of having a network defined in the CLI (see `sui client envs`). Preferably, we should keep things consistent across commands and simply have ZkLogin to simply use whatever network is currently set. Are you calling this from different scripts or why is it needed to supply a network argument here and not use the one that's already set in the CLI?

Reply:
@joyqvq any updates on this?

Reply:
i would prefer this to stay in the cli passed in args, bc the verify message doesnt go onchain or interact with the client in anyway. ppl may come to this request without needing to interact with the rest of the sui binary or keytool. it is purely used to determine what verifying key is used to verify the proof.